module/zfs/dmu.c

```diff
@@ -1196,7 +1196,7 @@ dmu_read_uio_dnode(dnode_t *dn, uio_t *uio, uint64_t size)
 
 #ifdef __FreeBSD__
 			err = vn_io_fault_uiomove((char *)db->db_data + bufoff,
-			    tocpy, uio);
+			    tocpy, GET_UIO_STRUCT(uio));
```

Comment:
This is one of those places where I'd suggest we create a new zfs uio wrapper function.  How about.

```c
int zfs_uio_fault_uiomove(char *, size_t, uio_rw_t, zfs_uio_t *);
```

Reply:
Same with this one.